test/models/learning_outcome_test.rb

```diff
@@ -0,0 +1,133 @@
+require 'test_helper'
+class LearningOutcomeTest < ActiveSupport::TestCase
+
+  def setup
+    data = {
+        code: 'COS10001',
+        name: 'Testing in Unit Tests',
+        description: 'Test unit',
+        # teaching_period: TeachingPeriod.find(3)
+      }
+    @unit = Unit.create(data)
+  end
+
+  # Check abbreviation uniqueness
+  def test_create_learning_outcome_with_unique_abbrev
+    #Create new unit.
+    learning_outcome_count = LearningOutcome.count
+    unit_1 = {
+        unit_id: 'ASD10001',
```

Comment:
Unit id is a relation to a unit - not the unit code. This wont work as you expect.

Try adding a simple test that you can actually create a learning outcome and that it is then available within the unit.